File: apps/client/types/task.ts
import z from "zod";

export const TaskSchema = z.object({
  id: z.uuid(),
  workspaceId: z.uuid(),
  name: z.string().nonempty(), // has to be unique within a workspace,
  createdAt: z.date(),
  updatedAt: z.date(),
});

export type Task = z.infer<typeof TaskSchema>;

File: apps/server/internal/models/task.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Task struct {
	ID          uuid.UUID `json:"id" db:"id"`
	WorkspaceID uuid.UUID `json:"workspaceId" db:"workspace_id"`
	Name        string    `json:"name" db:"name"`
	CreatedAt   time.Time `json:"createdAt" db:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt" db:"updated_at"`
}
